ecs-agent/logger/log.go

```diff
@@ -44,18 +44,34 @@ const (
 	DEFAULT_TIMESTAMP_FORMAT                 = time.RFC3339
 	DEFAULT_MAX_FILE_SIZE            float64 = 10
 	DEFAULT_MAX_ROLL_COUNT           int     = 24
+	DEFAULT_LOGTO_STDOUT                     = true
 )
 
+// Because timestamp format will be called in the custom formatter
+// for each log message processed, it should not be handled
+// with an explicitly write protected configuration.
+var timestampFormat = DEFAULT_TIMESTAMP_FORMAT
```

Comment:
Why do we need to move `timestampFormat` out of `logConfig`?

Reply:
timestampFormat is accessed each time we log something. Because Config is protected by a mutex, a data race can occur where we are trying to R/W at the same time. Two ways we can get around this are:

1. Mutex locking/unlocking Config during each log write(already done at the seelog level with a different mutex lock, this could also be slow)
2. Making timestampFormat independent of Config. The check will make sure timestampFormat is valid and allow quick access.